src/OrleansRuntime/Placement/StatelessWorkerDirector.cs

```diff
@@ -12,6 +12,26 @@ internal class StatelessWorkerDirector : IPlacementDirector<StatelessWorkerPlace
         public Task<PlacementResult> OnSelectActivation(
             PlacementStrategy strategy, GrainId target, IPlacementRuntime context)
         {
+            return Task.FromResult(SelectActivationCore(strategy, target, context));
+        }
+
+        public bool TrySelectActivationSynchronously(
+            PlacementStrategy strategy, GrainId target, IPlacementRuntime context, out PlacementResult placementResult)
+        {
+            placementResult = SelectActivationCore(strategy, target, context);
+            // stateless workers are always local, and  request for creation of a new activation will be created by PlacementDirectorsManager 
+            // in case of null returned by SelectActivationCore  
+            return true;
```

Comment:
Looked more carefully. I think you need to return false on null. 
SelectOrAddActivation will be triggered by null in the prev code, but in your new code if Try returns true, you never even call SelectOrAddActivation.

maybe its better to change Try to return the placement and not bool, to be similar to async select. it will make this kind of reasoning easier?